h/api/groups/auth.py

```diff
@@ -0,0 +1,50 @@
+def _authorized_for_group(effective_principals, group_hashid):
```

Comment:
It's supposed to test for the `group_hashid` being `None`. The idea is that the calling code just passes in `group_hashid=annotation.get('group')`, and the annotation may not contain any 'group' and the calling code doesn't need to worry about it.

I could have the calling code pass in the whole annotation instead of just the hashid? Make this code deal with how to get the annotation's group out of the annotation